suite/tests/client-interface/drmgr-test.dll.c

```diff
@@ -761,6 +795,21 @@ event_bb4_instru2instru(void *drcontext, void *tag, instrlist_t *bb, bool for_tr
     return DR_EMIT_DEFAULT;
 }
 
+static dr_emit_flags_t
+event_bb5_post_instru(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
+                      bool translating, void *user_data)
+{
+    CHECK(user_data == (void *)((ptr_uint_t)tag + 1), "user data not preserved");
```

Comment:
> but cannot test the counter in the event_exit

Instead of using a counter in `user_data`, maybe we can increment some global counter in each event? If there's some bug due to which the meta_instru callback isn't being invoked, it'd be good if the unit tests can detect that.

Reply:
Sure. Added a global counter and checked on exit.